packages/richtext-lexical/src/lexical/LexicalEditor.tsx

```diff
@@ -172,7 +222,9 @@ export const LexicalEditor: React.FC<
           </React.Fragment>
         )}
 
-        <TabIndentationPlugin />
+        {editorConfig.features.enabledFeatures.includes('indent') && tabIndentEnabled && (
+          <TabIndentationPlugin />
+        )}
```

Comment:
Actually, `<TabIndentationPlugin />` was never supposed to be in this file, but rather part of `packages/richtext-lexical/src/features/indent`.

The only change to make in this file is to remove this import from here. All the logic you're doing here with the escape key, and focus, should go into the indent feature. That way:
- The code is more modular and tidy.
- People who don't use the indent feature don't bloat their editor with code they don't use.